script.js

```diff
@@ -1,4 +1,943 @@
+                    Combined Numbers of each player: <br>${leaderBoardArray}
+                   <br><br>A new round will begin and all scores will reset on clicking "Submit" `;
+  } // else below is for catching error inputs.
+  else {
+    message2 = `${playerName}, please only:<br> 1) click the submit button, or<br> 2) type in "lowest", or, <br>3) type in "highest".`;
+  }
+
+  return gameLogicDisplay + message2;
+};
+
+//GLOBAL DEFINITIONS for MAIN FUCNTION
+var gameMode = "waiting for input";
+var player = "player 1";
+
+var main = function (input) {
+  var myOutputValue = "";
+  if (
+    gameMode == `waiting for input` &&
+    playerName == "" &&
+    (input.toLowerCase() == "highest" || input == "")
+  ) {
+    gameLogic = "highest";
+    myOutputValue = `Game Logic: ${gameLogic.toUpperCase()} combined number wins
+                    <br><br>Please enter the number of players:`;
+    gameMode = "getNumberOfPlayers";
+  } else if (
+    gameMode == `waiting for input` &&
+    playerName == "" &&
+    input.toLowerCase() == "lowest"
+  ) {
+    gameLogic = "lowest";
+    myOutputValue = `Game Logic: ${gameLogic.toUpperCase()} combined number wins
+                    <br><br>Please enter the number of players:`;
+    gameMode = "getNumberOfPlayers";
+    gameLogic = "lowest";
+  } else if (gameMode == `getNumberOfPlayers` && playerName == "") {
+    myOutputValue = getNumberOfPlayers(input);
+  } else if (gameMode == `inputQtyOfDice`) {
+    myOutputValue = inputQtyOfDice(input);
+  } else if (gameMode == `arrangeDice`) {
+    myOutputValue = arrangeDice(input);
+  }
+
+  return myOutputValue;
+}; */
+
+//==================================================================================================================================
+
+var Code04 =
+  " Beat That- with N Number of Dice, N number of Players and knockOut Mode";
+
+//Helper Functions
+var diceRoll = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  return randomInteger + 1;
+};
+
+//GLOBAL Definitions.
+var myOutputValue = "";
+var diceNumberRolled = 0;
+var numberOfPlayers = 0;
+var playerIndexNmbr = 0;
+var playerName = "";
+var gameCount = 0;
+var currentPlayerSumOfNumbers = 0;
+var gameLogic = "highest";
+var gameLogicDisplay = "";
+var qtyOfDice = 0;
+var leaderBoardArray = [];
+
+//Game Logic and Service parts
+
+var getNumberOfPlayers = function (qtyOfPlayers) {
+  // use over arcing "IF /ELSE " to control for highest or lowest mode and return error messages
+  if (qtyOfPlayers > 1 && isNaN(qtyOfPlayers) == false) {
+    numberOfPlayers = qtyOfPlayers;
+    var message0 = `Number of Players = ${qtyOfPlayers}
+                  <br><br>Please input how many dices to use: `;
+    gameMode = `inputQtyOfDice`;
+  } else {
+    message0 = `xxxx INPUT ERROR xxxx :<br>Please key in only numerals (> 1)`;
+    gameMode = "getNumberOfPlayers";
+  }
```

Comment:
You won't need line 599 since you only change the gamemode when the first if statement runs, but overall this is good